### PR TITLE
Save and load the user's /textcolor setting to persist it between game sessions

### PR DESCRIPTION
**Pull request by ZarothYe**
Add a new chronicle entry to store the RGB value of the user's current `/textcolor` setting.

### CODE REVIEW

File: Scripts/Python/ki/__init__.py [hunk 1]
```diff
@@ -1639,6 +1640,22 @@ def MessageHistory(self, control, set):
...
+    def DetermineTextColor(self):
+
+        vault = ptVault()
+        entry = vault.findChronicleEntry(kChronicleKITextColor)
+        if entry is not None:
+            colorStr = entry.chronicleGetValue()
+            PtDebugPrint(f"xKI.DetermineTextColor(): KI Text Color is: \"{colorStr}\".", level=kWarningLevel)
+            args = colorStr.split(",")
+            if len(args) == 3:
+                self.chatMgr.commandsProcessor.SetTextColor(args[0], args[1], args[2])
```
Comment (Hoikas) [inline Scripts/Python/ki/__init__.py:1657]:
```suggestion
                self.chatMgr.commandsProcessor.SetTextColor(*args)
```

You might also consider logging an error if the format is incorrect.

File: Scripts/Python/ki/xKIChat.py [hunk 2]
```diff
@@ -1217,6 +1217,23 @@ def ChangePassword(self, newPassword):
...
+        newVal = ""
+        if isinstance(self.chatMgr.chatTextColor, ptColor):
+            newVal = f"{self.chatMgr.chatTextColor.getRed()},{self.chatMgr.chatTextColor.getGreen()},{self.chatMgr.chatTextColor.getBlue()}"
+
+        PtDebugPrint(f"xKIChat.UpdateTextColorChronicle(): Setting KI Text Color chronicle to: \"{newVal}\".", level=kWarningLevel)
+        if entry is not None:
+            entry.chronicleSetValue(newVal)
+            entry.save()
+        else:
+            vault.addChronicleEntry(kChronicleKITextColor, kChronicleKITextColorType, newVal)
```
Comment (Hoikas) [inline Scripts/Python/ki/xKIChat.py:1235]:
```suggestion
        newVal = ""
        if isinstance(self.chatMgr.chatTextColor, ptColor):
            newVal = f"{self.chatMgr.chatTextColor.getRed()},{self.chatMgr.chatTextColor.getGreen()},{self.chatMgr.chatTextColor.getBlue()}"

        PtDebugPrint(f"xKIChat.UpdateTextColorChronicle(): Setting KI Text Color chronicle to: \"{newVal}\".", level=kWarningLevel)
        vault.addChronicleEntry(kChronicleKITextColor, kChronicleKITextColorType, newVal)
```

`addChronicleEntry()` updates any already existing entry.

Comment (ZarothYe) [reply Scripts/Python/ki/xKIChat.py:1235]: Ooooooh... I, uh, kind of copied that if/else pattern from somewhere else so I didn't know that. Apparently that usage is poor in a lot of places. 😬

Comment (Hoikas) [reply Scripts/Python/ki/xKIChat.py:1235]: It's possible that some older variant of Uru did *not* do that check, but MOUL does.

**Review by Hoikas — APPROVED** [review]

File: Scripts/Python/ki/xKIChat.py [hunk 3]
```diff
@@ -1217,6 +1217,19 @@ def ChangePassword(self, newPassword):
             return
         PtChangePassword(newPassword)
 
+    ## Update the Chronicle's KI Text Color values.
+    # This uses an RGB triplet with parts separated by "," or an empty string for the default chat color.
+    def UpdateTextColorChronicle(self):
+
+        vault = ptVault()
+        entry = vault.findChronicleEntry(kChronicleKITextColor)
```
Comment (Hoikas) [inline Scripts/Python/ki/xKIChat.py:1225]:
```suggestion
```

This seems to be unused.

**Review by Hoikas — COMMENTED** [review]
Just noticed one straggler.

**Review by Hoikas — APPROVED** [review]

**Pull request merged**